app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java

```diff
@@ -77,64 +72,54 @@ public void onCreate(Bundle savedInstanceState)
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState)
-    {
+                             Bundle savedInstanceState) {
 
-        View view = inflater.inflate(R.layout.fragment_speak, container, false);
+        View view = inflater.inflate(R.layout.fragment_speech, container, false);
         ButterKnife.bind(this, view);
-        listener = new SpeechListener(new SpeechCallback()
-        {
+        listener = new SpeechListener(new SpeechCallback() {
             @Override
-            public void callback(ArrayList voiceResult)
-            {
+            public void callback(ArrayList voiceResult) {
                 voiceInput = voiceResult;
                 speechTextDisplay.setText(voiceInput.get(0).toString());
                 isListening = false;
+                Log.d(DEBUG_TAG, "Got result, stopped listening.");
             }
 
             @Override
-            public void errorCallback(int errorCode)
-            {
-                if (errorCode == SpeechRecognizer.ERROR_NO_MATCH) {
+            public void errorCallback(int errorCode, int numErrors) {
+                isListening = false;
+                Log.d(DEBUG_TAG, "Got error, stopped listening.");
+                if (errorCode == SpeechRecognizer.ERROR_NO_MATCH && numErrors == 1) { // error 7
+                    //numErrors is to check to make sure this is first time you get the error so you
+                    //don't get a bunch of toasts from unrecognized partial results
                     //TODO: change this to saying out loud, "please try again"
                     Toast.makeText(getActivity().getApplicationContext(),
                             "Error: Speech was not recognized.", Toast.LENGTH_SHORT).show();
-                }
-                if (errorCode == SpeechRecognizer.ERROR_SPEECH_TIMEOUT) {
+                } else if (errorCode == SpeechRecognizer.ERROR_SPEECH_TIMEOUT) { //error 6
```

Comment:
I don't think it really matters since you'll only get one error if theres a speech timeout (it's either the timeout or recognition)